pg_bm25/src/parade_index/index.rs

```diff
@@ -108,22 +110,28 @@ impl ParadeIndex {
 
     pub fn bulk_delete(
         &self,
-        stats_binding: *mut IndexBulkDeleteResult,
+        mut stats_binding: PgBox<IndexBulkDeleteResult, AllocatedByPostgres>,
         callback: IndexBulkDeleteCallback,
         callback_state: *mut ::std::os::raw::c_void,
-    ) {
-        let mut index_writer = self.underlying_index.writer(50_000_000).unwrap(); // Adjust the size as necessary
+    ) -> PgBox<IndexBulkDeleteResult, AllocatedByPostgres> {
+        let mut index_writer = self
+            .underlying_index
+            .writer(INDEX_TANTIVY_MEMORY_BUDGET)
+            .unwrap(); // Adjust the size as
+                       // necessary
 
         let schema = self.underlying_index.schema();
         let heap_tid_field = schema
             .get_field("heap_tid")
             .expect("Field 'heap_tid' not found in schema");
 
-        let searcher = self
+        let reader = self
             .underlying_index
-            .reader()
-            .expect("Failed to acquire index reader")
-            .searcher();
+            .reader_builder()
+            .reload_policy(tantivy::ReloadPolicy::OnCommit)
+            .try_into();
+
+        let searcher = reader.expect("could not create index").searcher();
```

Comment:
We recently added a `searcher()` method on `ParadeIndex` to retrieve a new `Searcher` instance. It takes advantage of new caching to avoid re-creating the reader. Could you use that here when you rebase?

Reply:
@neilyio you'll need to please check on this with @rebasedming, and probably take over this PR to do these changes.
I'm unfortunately not working anymore on it going forward. thanks!

Reply:
Sounds good, thank you for help so far @aragalie!